std/assembly/string.ts

```diff
@@ -380,6 +386,36 @@ export class String {
     );
     return out;
   }
+
+  repeat(count: i32 = 0): String {
+    assert(this !== null);
+    var length = this.length;
+
+    // Most browsers can't handle strings 1 << 28 chars or longer
+    if (count < 0 || length * count > (1 << 28)) {
+      throw new RangeError("Invalid count value");
+    }
+
+    if (count === 0 || !length) return EMPTY;
+    if (count === 1) return this;
+
+    var result = allocate(length * count);
+    var strLen = length << 1;
+
+    for (let offset = 0, len = strLen * count; offset < len; offset += strLen) {
+      move_memory(
```

Comment:
I guess copy size could also increase. Like, repeating "a" four times would first create "aa" and then copy "aa" to "aaaa". Wdyt?